src/shared/utilities/promptSetting.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../logger'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+export type PROMPT_NAME =
+    | 'warnBeforeRunCommand'
+    | 'warnBeforeEnablingExcecuteCommand'
+    | 'warnBeforeDisablingExecuteCommand'
+export const DEFAULT_PROMPTS: PromptSetting = {
+    warnBeforeRunCommand: true,
+    warnBeforeEnablingExcecuteCommand: true,
+    warnBeforeDisablingExecuteCommand: true,
+}
+
+export interface PromptSetting {
+    [promptName: string]: boolean
+}
+
+export function disablePrompt(promptName: PROMPT_NAME, settings: SettingsConfiguration): void {
+    try {
+        const prompts = settings.readSetting<PromptSetting>('togglePrompts', DEFAULT_PROMPTS)
+
+        prompts[promptName] = false
+        settings.writeSetting('togglePrompts', prompts, vscode.ConfigurationTarget.Global)
```

Comment:
we can avoid unnecessary updates to the user's settings file: before doing `prompts[promptName] = false`, check if we are actually changing anything.

```
if (!prompts.includes(promptName) || prompts[promptName] !== false) {
    settings.writeSetting(...)
}
```